gateway/cert.go

```diff
@@ -389,7 +390,20 @@ func getTLSConfigForClient(baseConfig *tls.Config, listenPort int) func(hello *t
 			}
 		}
 
-		newConfig.ClientAuth = domainRequireCert[hello.ServerName]
+		matches := false
+		for key, clientAuth := range domainRequireCert {
+			req := http.Request{Host: hello.ServerName, URL: &url.URL{}}
+			if mux.NewRouter().Host(key).Match(&req, &mux.RouteMatch{}) {
+				newConfig.ClientAuth = clientAuth
+				matches = true
+				break
+			}
+		}
+
+		if !matches {
```

Comment:
We can remove negative logic, and `matches` boolean var too, and just move this to line 393, e.g. always set to `tls.NoClientCert` by default.